bip32.h

```diff
@@ -45,7 +45,12 @@ typedef struct {
 	uint32_t depth;
 	uint32_t child_num;
 	uint8_t chain_code[32];
+
+	// the private key and private key extension must be in this order and next to each other
+	// because some of the ed25519 functions works with 32-byte and some with 64-byte private key
 	uint8_t private_key[32];
+	uint8_t private_key_extension[32];
```

Comment:
TBH I don't really like this. Most of the code check analysis tools will scream anyway when you'll try to write 64 bytes into private_key field.

I suggest to move `private_key_extension` AFTER `public_key` field. Then fix all broken stuff (it should be quite obvious now). Then return it back, but without the comment.

From what I can tell fixing the issues should be quite easy, just split 

`memcpy(priv_key,inout->private_key,32);` into

```
memcpy(priv_key,inout->private_key,64);
memcpy(priv_key+32,inout->private_key_extension,32);
```

and

`memcpy(inout->private_key, res_key, 64);` into

```
memcpy(inout->private_key, res_key, 32);
memcpy(inout->private_key_extension, res_key + 32, 32);
```

+ maybe couple of more similar things.